pkg/controllers/state/cluster.go

```diff
@@ -90,6 +103,14 @@ func (c *Cluster) Synced(ctx context.Context) (synced bool) {
 	// Set the metric to whatever the result of the Synced() call is
 	defer func() {
 		clusterStateSynced.Set(lo.Ternary[float64](synced, 1, 0))
+		if synced {
+			c.lastSynced = c.clock.Now()
+		}
+		// If we have been out of sync for a while, and we haven't logged about it recently, log it
+		if time.Since(c.lastSynced) > AbnormalClusterSyncDuration && c.cm.HasChanged("ClusterSynced", synced) {
```

Comment:
Do we add unit tests for testing logging changes in the project?

Reply:
No, feels a little overkill to add it as well. I think just testing that you get the behavior that you are expecting here should be sufficient